File: test/org/cerion/symcalc/expression/function/core/NTest.java
package org.cerion.symcalc.expression.function.core;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.constant.Pi;
import org.cerion.symcalc.expression.function.arithmetic.Plus;
import org.cerion.symcalc.expression.number.IntegerNum;
import org.cerion.symcalc.expression.number.RealNum;
import org.junit.Test;

import static org.junit.Assert.*;

public class NTest {

    @Test
    public void constantPlusInteger() {
        Expr e = new N(new Plus(new Pi(), new IntegerNum(5)));
        assertEquals(RealNum.Companion.create(8.141592653589793), e.eval());

        // TODO this is what mathematica does but it doesn't seem right
        e = new Plus(new N(new IntegerNum(5)), new Pi());
        assertEquals(RealNum.Companion.create(8.141592653589793), e.eval());
    }
}

File: src/test/java/function/core/NTest.kt
package org.cerion.symcalc.function.core

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.constant.Pi
import org.cerion.symcalc.function.arithmetic.Plus
import org.cerion.symcalc.expression.number.Integer
import org.cerion.symcalc.expression.number.Rational
import org.cerion.symcalc.expression.number.RealBigDec
import org.cerion.symcalc.expression.number.RealDouble
import kotlin.test.Test
import kotlin.test.assertEquals

class NTest {

    @Test
    fun constantPlusInteger() {
        var e: Expr = N(Plus(Pi(), Integer(5)))
        assertEquals(RealDouble(8.141592653589793), e.eval())

        e = Plus(N(Integer(5)), Pi())
        assertEquals(RealDouble(8.141592653589793), e.eval())
    }

    @Test
    fun zeroPrecision() {
        assertEquals(RealBigDec("0.0"), N(Rational(1,3), Integer(0)).eval())
        assertEquals(RealBigDec("0.0"), N(Pi(), Integer(0)).eval())
    }
}